core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -1125,6 +1125,78 @@ public boolean hasNext() {
     return Streams.stream(rollingIterator(n));
   }
 
+  /**
+   * Transposes data in the table, switching rows for columns.
+   * For example, a table like this.
+   *  label  |  value1  |  value2  |
+   * -------------------------------
+   *   row1  |       1  |       2  |
+   *   row2  |     1.1  |     2.1  |
+   *   row3  |     1.2  |     2.2  |
+   *
+   *  Is transposed into the following
+   *  label   |  row1  |  row2  |  row3  |
+   * -------------------------------------
+   *  value1  |     1  |   1.1  |   1.2  |
+   *  value2  |     2  |   2.1  |   2.2  |
+   *
+   * @return The transposed table
+   */
+  public Table transpose() {
+    Table transposed = Table.create(this.name);
+    Column firstColumn = this.column(0);
+
+    ColumnType[] types = this.columnTypes();
+    //If all columns are of the same type
+    long distinctColumnTypesCount = Arrays.stream(types).skip(1).distinct().count();
+
+    if (distinctColumnTypesCount == 1) {
```

Comment:
I'd reverse this check:

```
if (distinctColumnTypesCount > 1) {
      throw new IllegalArgumentException(
          "Transpose currently only works where value columns are of the same type");
}
```

Then you can remove a level of indentation from the rest of the code

Reply:
Yep, agree will change